app/javascript/components/mentoring/automation/RepresentationList.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react'
+import { Pagination } from '../../common/Pagination'
+import { FetchingBoundary } from '../../FetchingBoundary'
+import { APIResponse } from './useMentoringAutomation'
+import { QueryStatus } from 'react-query'
+import { AutomationListElement } from './AutomationListElement'
+import { GraphicalIcon } from '../../common'
+
+const DEFAULT_ERROR = new Error('Unable to fetch queue')
+
+type Props = {
+  resolvedData: APIResponse | undefined
+  latestData: APIResponse | undefined
+  page: number
+  setPage: (page: number) => void
+  withFeedback: boolean
+}
+
+export const RepresentationList = ({
+  status,
+  error,
+  ...props
+}: { status: QueryStatus; error: unknown } & Props): JSX.Element => {
+  return (
+    <FetchingBoundary
+      status={status}
+      error={error}
+      defaultError={DEFAULT_ERROR}
+    >
+      <Component {...props} />
+    </FetchingBoundary>
+  )
+}
+
+function Component({
+  resolvedData,
+  latestData,
+  page,
+  setPage,
+  withFeedback,
+}: Props) {
+  return (
```

Comment:
How about you do 
```if(!resolvedData || !resolvedData.results) { return <></> }```

as a guard first, then the code below can be indented a couple of layers less.